shared/storage.ts

```diff
@@ -79,11 +80,34 @@ class Storage {
 
       return await MMKV.getItem(key);
     } catch (error) {
+      const isVCKey = VCMetadata.isVCKey(key);
+
+      if (isVCKey) {
+        const isDownloaded = await this.isVCAlreadyDownloaded(
+          key,
+          encryptionKey,
+        );
+
+        if (isDownloaded && error.message.includes(ENOENT)) {
+          throw new Error(ENOENT);
+        }
+        console.log('Error Occurred while retriving from Storage.', error);
+        throw error;
+      }
+
       console.log('Error Occurred while retriving from Storage.', error);
       throw error;
     }
   };
 
+  private static async isVCAlreadyDownloaded(
+    key: string,
+    encryptionKey: string,
+  ) {
+    const storedHMACofCurrentVC = await this.readHmacForVC(key, encryptionKey);
+    return storedHMACofCurrentVC !== null;
```

Comment:
if both the file and hmac is deleted by attacker, won't this put our app in a state where vc card is stuck at loading state